conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -1424,3 +1424,117 @@ def package(self):
+        if(WIN32)
+            set(MYTOOL_SCRIPT "mytool.bat")
+        else()
+            set(MYTOOL_SCRIPT "mytool.sh")
+        endif()
+        add_custom_target(run_mytool COMMAND ${MYTOOL_SCRIPT})
+        # build var should be available at configure
+        set(MY_BUILD_VAR $ENV{MY_BUILD_VAR})
+        if (MY_BUILD_VAR)
+            message("MY_BUILD_VAR:${MY_BUILD_VAR}")
+        else()
+            message("MY_BUILD_VAR NOT FOUND")
+        endif()
+        # run var should not be available at configure, just when testing
+        set(MY_RUNVAR $ENV{MY_RUNVAR})
+        if (MY_RUNVAR)
+            message("MY_RUNVAR:${MY_RUNVAR}")
+        else()
+            message("MY_RUNVAR NOT FOUND")
+        endif()
+        enable_testing()
+        add_executable(test_env test_env.cpp)
+        add_test(NAME TestRunEnv COMMAND test_env)
+    """)
+
+    c.save({"tool.py": tool.format(""),
+            "test_tool.py": tool.format('self.runenv_info.define("MY_RUNVAR", "MY_RUNVAR_VALUE")'),
+            "conanfile.txt": consumer,
+            "CMakeLists.txt": cmakelists,
+            "test_env.cpp": test_env})
+
+    c.run("create tool.py --name=mytool")
+
+    c.run("create test_tool.py --name=mytesttool")
+    c.run("create test_tool.py --name=mytesttool -s build_type=Debug")
+
+    # do a first conan install with env disabled just to test that the conf works
+    c.run("install . -g CMakeToolchain -g CMakeDeps -c tools.cmake.cmaketoolchain:presets_environment=disabled")
+
+    presets_path = os.path.join("build", "Release", "generators", "CMakePresets.json") \
+        if platform.system() != "Windows" else os.path.join("build", "generators", "CMakePresets.json")
+    presets = json.loads(c.load(presets_path))
+
+    assert presets["configurePresets"][0].get("env") is None
+
+    c.run("install . -g CMakeToolchain -g CMakeDeps")
+    c.run("install . -g CMakeToolchain -g CMakeDeps -s:h build_type=Debug")
+
+    # test that the buildenv is correctly injected to configure and build steps
+    # that the runenv is not injected to configure, but it is when running tests
+
+    preset = "conan-default" if platform.system() == "Windows" else "conan-release"
+
+    c.run_command(f"cmake --preset {preset}")
+    assert "MY_BUILD_VAR:MY_BUILDVAR_VALUE" in c.out
+    assert "MY_RUNVAR NOT FOUND" in c.out
+    c.run_command("cmake --build --preset conan-release --target run_mytool --target test_env")
+    assert "running: mytool/0.1" in c.out
+
+    c.run_command("ctest --preset conan-release")
+    assert "tests passed" in c.out
+
+    if platform.system() != "Windows":
+        c.run_command("cmake --preset conan-debug")
+        assert "MY_BUILD_VAR:MY_BUILDVAR_VALUE" in c.out
+        assert "MY_RUNVAR NOT FOUND" in c.out
+
+    c.run_command("cmake --build --preset conan-debug --target run_mytool --target test_env")
+    assert "running: mytool/0.1" in c.out
+
+    c.run_command("ctest --preset conan-debug")
+    assert "tests passed" in c.out
```

Comment:
It reads fine, but IMO it'd be great if we could split this test into three different ones:
* one for conf
* one for buildenv/configure
* one for runenv/tests

Reply:
Maybe we can keep this for the moment, as the PR is already green, and probably refactor when we add more tests in the future if we extend this feature?